### ISSUE TITLE
Scaled Object fails to find a RabbitMQ queue after working the 1st time it's used in Release 2.5.0

### ISSUE THREAD

**Issue by scubakiz** [kedacore/keda#2505]
### Report

I've been using this same ScaledObject for over 6 months with no issues:

```
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: keda-rabbit-so  
spec: 
  scaleTargetRef:
    name: queue-processer-dep  
  triggers:
  - type: rabbitmq
    metadata:
      protocol: amqp
      queueName: SampleQueue
      mode: QueueLength
      value: "10"      
      host: amqp://guest:guest@rabbit-svc.default.svc.cluster.local:5672
...
```
After installing the latest version (2.5.0) in a new cluster, the object works as expected.

I uninstalled the Helm chart with the ScaledObject and then reinstalled it a few hours later.  With nothing changed in my chart, the ScaledObject can longer find the queue the second time.

```
Warning	FailedGetExternalMetric	32s	horizontal-pod-autoscaler	unable to get external metric default/s0-rabbitmq-SampleQueue/&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: No matching metrics found for s0-rabbitmq-samplequeue
Warning	FailedComputeMetricsReplicas	32s	horizontal-pod-autoscaler	invalid metrics (1 invalid out of 1), first error is: failed to get s0-rabbitmq-SampleQueue external metric: unable to get external metric default/s0-rabbitmq-SampleQueue/&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: No matching metrics found for s0-rabbitmq-samplequeue
```

Repeated uninstalling/installing the Helm chart produces the same result.  If the Keda chart is uninstalled and reinstalled, everything works again, but only the first time.  

Uninstalling Keda 2.5.0 and installing Keda 2.4.0 fixes the problem and the ScaledObject works as expected every time.  

I noticed that the queue naming convention seems to be different in 2.4.0 compared with 2.5.0.  Is there a new naming requirement for queues (for example, they must be lower case)?  If so, why does it work the first time?

### Expected Behavior

The ScaledObject should work correctly every time.

### Actual Behavior

In Release 2.5.0, the ScaledObject works the first time, but fails all subsequent times.

### Steps to Reproduce the Problem

In 2.5.0
1. Deploy the ScaledObject - works as expected
2. Delete the ScaledObject 
3. Deploy the ScaledObject - fails
4. Delete the ScaledObject 
5. Deploy the ScaledObject - fails

In 2.4.0
1. Deploy the ScaledObject - works as expected
2. Delete the ScaledObject 
3. Deploy the ScaledObject - works as expected
4. Delete the ScaledObject 
5. Deploy the ScaledObject - works as expected

### Logs from KEDA operator

Release 2.5.0 - First run:

```
Normal	SuccessfulRescale	4m37s	horizontal-pod-autoscaler	New size: 11; reason: external metric s0-rabbitmq-SampleQueue(&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}) above target
Normal	SuccessfulRescale	3m22s	horizontal-pod-autoscaler	New size: 22; reason: external metric s0-rabbitmq-SampleQueue(&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}) above target
Normal	SuccessfulRescale	3m7s	horizontal-pod-autoscaler	New size: 30; reason: external metric s0-rabbitmq-SampleQueue(&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}) above target
```

Release 2.5.0 - Subsequent runs:
```
Warning	FailedGetExternalMetric	32s	horizontal-pod-autoscaler	unable to get external metric default/s0-rabbitmq-SampleQueue/&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: No matching metrics found for s0-rabbitmq-samplequeue
Warning	FailedComputeMetricsReplicas	32s	horizontal-pod-autoscaler	invalid metrics (1 invalid out of 1), first error is: failed to get s0-rabbitmq-SampleQueue external metric: unable to get external metric default/s0-rabbitmq-SampleQueue/&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: No matching metrics found for s0-rabbitmq-samplequeue
```

In Release 2.4.0 - Every run:
```
Normal	SuccessfulRescale	34s	horizontal-pod-autoscaler	New size: 11; reason: external metric rabbitmq-SampleQueue(&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}) above target
Normal	SuccessfulRescale	19s	horizontal-pod-autoscaler	New size: 22; reason: external metric rabbitmq-SampleQueue(&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}) above target
Normal	SuccessfulRescale	4s	horizontal-pod-autoscaler	New size: 30; reason: external metric rabbitmq-SampleQueue(&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: keda-rabbit-so,},MatchExpressions:[]LabelSelectorRequirement{},}) above target
```

### KEDA Version

2.5.0

### Kubernetes Version

1.21

### Platform

Microsoft Azure

### Scaler Details

RabbitMQ

### Anything else?

It seems that something is cached in the Keda operator.  I changed the name of the queue and tried it again, but the errors still listed the original queue name.

**Comment by zroubalik** [comment]
Hi you have probably hit an issue, that was introduced in 2.5.0 and has been already fixed and will be released in 2.6.0

https://github.com/kedacore/keda/pull/2408

**Comment by tomkerkhove** [comment]
Closing as duplicate
